plugins/extra/metar.go

```diff
@@ -73,3 +79,61 @@ func Metar(code string) (string, error) {
 
 	return "", errors.New("No results")
 }
+
+func tafCallback(b *seabird.Bot, m *irc.Message) {
+	if !m.FromChannel() {
+		return
+	}
+
+	r, err := TAF(m.Trailing())
+	if err != nil {
+		b.MentionReply(m, "Error: %s", err)
+		return
+	}
+
+	b.MentionReply(m, "%s", r)
+}
+
+// TAF is a simple function which takes a string representing the
+// airport code and returns a string representing the response or an
+// error.
+func TAF(code string) (string, error) {
+	code = strings.ToUpper(code)
+
+	for _, letter := range code {
+		if !unicode.IsDigit(letter) && !unicode.IsLetter(letter) {
+			return "", errors.New("Not a valid airport code")
+		}
+	}
+
+	resp, err := http.Get(fmt.Sprintf("http://tgftp.nws.noaa.gov/data/forecasts/taf/stations/%s.TXT", code))
+	if err != nil {
+		return "", errors.New("NOAA appears to be down")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return "", errors.New("Station does not exist")
```

Comment:
nit: maybe `resp.StatusCode != 200 && resp.StatusCode >= 400 && resp.StatusCode < 500 {`

Reply:
This is the same check we use for metar. It's somewhat lazy... but eh.